mlx/random.cpp

```diff
@@ -136,7 +136,7 @@ array uniform(
   auto out = bits(shape, size_of(dtype), key, stream);
   out = astype(divide(out, maxval, stream), dtype, stream);
   out = minimum(out, upper, stream);
-  return add(multiply(range, out, stream), low, stream);
+  return astype(add(multiply(range, out, stream), low, stream), dtype, stream);
```

Comment:
Great catch! I would recommend we cast the range to the type. By this point `out` should have the right type already so if we cast `range` either on L106 directly or above this line, then the remaining computation will be done in the requested precision.

Reply:
this won't be sufficient, as `low` will still be float32. 
It depends on whether you want to compute in lower precision or compute in higher precision and then round to lower. 
If I understand you right, you want to compute in lower precision (which probably makes sense to avoid any intermediate big arrays). 
Then I'd suggest to cast `low` and `high` in the beginning, which will also make `range` the right format. 
Check on the updated code...

Reply:
Yea that was my point, to avoid materializing bigger arrays when using lower precision. And yes good point regarding low. Casting low/high is the right call

Reply:
Thank you for your always positive and encouraging feedback and quick responses.